modules/order/src/EventSubscriber/OrderMailer.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Drupal\commerce_order\EventSubscriber;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Mail\MailManagerInterface;
+use Drupal\Core\Render\Markup;
+use Drupal\Core\Render\RenderContext;
+use Drupal\Core\Render\Renderer;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\state_machine\Event\WorkflowTransitionEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Listens to order transition events and sends emails.
+ */
+class OrderMailer implements EventSubscriberInterface {
```

Comment:
This is an event subscriber, and core usually adds Subscriber to such class names.
Maybe OrderReceiptSubscriber makes the most sense? or OrderMailSubscriber, but we're calling it the "receipt" in the order type.